### ISSUE TITLE
Disable manual health check for control & hybrid nodes

### ISSUE THREAD

**Issue by AlanCoding** [ansible/awx#12966]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Feature type

Enhancement to Existing Feature

### Feature Summary

We have an endpoint `/api/v2/instances/N/health_check/`, and this gives a 404 for hop nodes.

The link is also not shown in related links for hop nodes.

This issue argues that these should apply to control & hybrid nodes as well. They run health checks on a periodic basis so there is no need for the user to manually initiate it.

In other words, manually health checks should only be possible for execution nodes.

### Select the relevant components

- [ ] UI
- [X] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [ ] Other

### Steps to reproduce

If you POST to that endpoint for these node types

### Current results

Right now it will submit a task to run a health check, but expectations are a little unclear, since the result is likely to come from the 20 second periodic task.

### Sugested feature result

The API will give a 404 for this endpoint for those instance types, and the UI will hide the button.

The health check button will only be shown for execution nodes.

**Comment by AlexSCorey** [comment]
I believe this is already addressed in the UI

**Comment by AlanCoding** [comment]
@AlexSCorey can this issue be closed now? Your PR addressed both the UI and the API.
